xbmc/linux/XMemUtils.cpp

```diff
@@ -95,9 +95,16 @@ void GlobalMemoryStatusEx(LPMEMORYSTATUSEX lpBuffer)
   {
       // Find page size.
       int pageSize;
+#if defined(__aarch64__)
+      // Workaround for 64-bit arm devices (e.g. iPhone 5s, etc.).
+      // sysctl() and host_page_size() give both wrong 4x value (16384).
+      // So, we need to hardcode page size to 4096 here.
+      pageSize = 4096;
+#else
```

Comment:
yes, this is a darwin only bug, needs rechecking if iOS/tvOS 9 and above is fixed.